heareval/tasks/pipeline.py

```diff
@@ -286,37 +271,65 @@ def split_train_test_val(self, metadata: pd.DataFrame):
             + f"now sampled with split key are: {splits_to_sample}"
         )
 
-        # Depending on whether valid and test are already present, the percentage can
-        # either be the predefined percentage or 0
-        validation_percentage = (
-            VALIDATION_PERCENTAGE if "valid" in splits_to_sample else 0
-        )
-        test_percentage = TEST_PERCENTAGE if "test" in splits_to_sample else 0
-
-        metadata.reset_index(drop=True, inplace=True)
-        metadata[metadata["split"] == "train"] = metadata[
-            metadata["split"] == "train"
-        ].assign(
-            split=lambda df: df["split_key"].apply(
-                # Use the which set to split the train into the required splits
-                lambda split_key: which_set(
-                    split_key,
-                    validation_percentage=validation_percentage,
-                    test_percentage=test_percentage,
-                )
+        train_percentage: float
+        valid_percentage: float
+        test_percentage: float
+
+        # If we want a 60/20/20 split, but we already have test
+        # then we want to do a 75/25/0 split so that train is still 3x validation
+        if splits_to_sample == set():
+            return metadata
+        elif splits_to_sample == set("valid"):
+            tot = TRAIN_PERCENTAGE + TEST_PERCENTAGE
+            train_percentage = (
+                TRAIN_PERCENTAGE + TRAIN_PERCENTAGE * VALIDATION_PERCENTAGE / tot
             )
-        )
+            valid_percentage = 0
```

Comment:
I'm confused, if the split we want to sample is "valid", then why do we set the percentage to zero?

Reply:
I find this calculation of percentages a bit hard to follow. Can this be simplified?

When you need to sample validation from training:
```
tot  = TRAIN_PERCENTAGE + VALIDATION_PERCENTAGE
train_percentage = TRAIN_PERCENTAGE + TRAIN_PERCENTAGE * VALIDATION_PERCENTAGE / tot
validation_percentage = 100.0 - train_percentage
test_percentage = 0.0
```

Reply:
Unless I'm missing something.

Reply:
"I'm confused, if the split we want to sample is "valid", then why do we set the percentage to zero?" My mistake, I'm fixing

Reply:
@jorshi check now?